### ISSUE TITLE
allow spice.so to be found from env var

### ISSUE THREAD

**Issue by AndrewAnnex** [AndrewAnnex/SpiceyPy#349]
currently spiceypy will on pip install, download and build spice.so for users but some users are offline and would like to be able to install from source with a user provided env variable that points to the shared library file, getspice or setup.py should be updated to look for some env flag and to skip the steps for building spice, libspice.py would need to updated to point there optionally

**Comment by jessemapel** [comment]
Thoughts on environment variable name? SPICE_LIB_PATH?

**Comment by AndrewAnnex** [comment]
yeah that would probably be sufficient, I think it is more common to point to the directory and then expect there to be a file within the directory rather than defining the full path to the file. Alternatively this env var could point to a folder of cspice that has already been downloaded and then the get spice building scripts could operate there instead of attempting to download the source code for spice.... maybe this is really two different scenarios

**Comment by jessemapel** [comment]
I think you could reasonably handle both cases. First search the base directory for the binary, if you don't find it then try and search like it's a cspice install.

**Comment by AndrewAnnex** [comment]
note some of these issues are intended for GSoC

**Comment by medley56** [comment]
We are trying to solve this problem for a system that (for an undetermined reason) absolutely refuses to download cspice from NAIF during pip install (it has TLS 1.2). 

Instead of pointing an environment variable, I suggest modifying `get_spice.py` to search in a specific location for the cspice source. If it doesn't find it, then it downloads it. In fact, it looks like this could be added as an `else` in `InstallCSpice.check_for_spice()`.

**Comment by AndrewAnnex** [comment]
@medley56 that was a fallback that I was also considering and could be easily implemented alongside this change. At the moment I am pretty busy so I hope that someone would be willing to implement the functionality as a contribution, as it would all be a fairly easy thing to do using pathlib / os.environ .

**Comment by medley56** [comment]
So I just implemented this in a rudimentary way by changing like 3 lines of code. It's not particularly safe or rubust though. 

```
258     def check_for_spice():
259         print("Checking the path", cspice_dir)
260         if os.path.exists(cspice_dir):
261             print("Found spice at {0}. Skipping download of CSPICE.".format(cspice_dir))
262             return True
263         else:
264             raise NotImplementedError
265             message = "Unable to find CSPICE at {0}. Attempting to Download CSPICE For you:".format(
266                 cspice_dir
267             )
268             print(message)
269             # Download cspice using getspice.py
270             GetCSPICE(version="N0066")
271             if not os.path.exists(cspice_dir):
272                 message = "Unable to find CSPICE at {0}. Exiting".format(cspice_dir)
273                 sys.exit(message)
274             return True
275         return False
```

**Comment by AndrewAnnex** [comment]
Great would you be interested in submitting a PR with the changes, I can review it later tonight or the weekend and clean it up for you if you want (just enable maintainer edits to the pull request). I’ll want to add tests using the CI services which would take a bit more thought 

-Andrew Annex

**Comment by AndrewAnnex** [comment]
It should probably be updated to use some environment  variable, maybe "CSPICE_SRC_PATH", if that variable is defined, check if the source is at that location and build. if "CSPICE_SHAREDLIB" is defined then skip the other steps and use that shared library file instead

**Comment by medley56** [comment]
My PR addresses the usecase where you already have the cspice.tar.Z downloaded (e.g. if you don't want to deal with updating SSL). It doesn't address the shared library usecase. The environment variable I used was SPICEYPY_CSPICE_PATH.

**Comment by AndrewAnnex** [comment]
closed by #371
